lib/mongo_ecto.ex

```diff
@@ -407,23 +407,68 @@ defmodule Mongo.Ecto do
   end
 
   @doc false
-  def load(:binary_id, data),
-    do: Ecto.Type.load(ObjectID, data, &load/2)
-  def load(Ecto.Date, {date, _time}),
-    do: Ecto.Type.load(Ecto.Date, date, &dump/2)
-  def load(:map, []),
-    do: {:ok, %{}}
+  def load(_type, nil),
+    do: {:ok, nil}
+  # wat
+  def load(:binary_id, <<_::binary-12>> = binary),
+    do: {:ok, ObjectID.encode(binary)}
```

Comment:
Certainly not from Ecto. From here I think: https://github.com/michalmuskala/mongodb_ecto/blob/02fe793d04e9c7833259466ba5dae8b58d755b7d/lib/mongo_ecto.ex#L475-L484